### DISCUSSION TITLE
Extensions specified in `devcontainer.json` file do not install in the correct extension host

### DISCUSSION THREAD

**Discussion by thegecko** [community/community#41636] [Codespaces]
### Select Topic Area

Bug

### Body

As outlined over at:

https://github.com/microsoft/vscode/issues/154563

VS Code web extensions specified in a `devcontainer.json` file on codespaces do not install in the correct extension host. From that issue:

- VS Code Version: 1.69.0 (codespaces)
- OS Version: MacOS 12.4 (chrome)

Steps to Reproduce:

1. Use/identify an extension which can run in different extension hosts depending on host availability following the pattern agreed in https://github.com/microsoft/vscode/issues/141322 (e.g. [arm.device-manager](https://marketplace.visualstudio.com/items?itemName=Arm.device-manager))
2. Create a GitHub repository containing a `devcontainer.json` file with this extension (e.g. [this one](https://github.com/thegecko/getting-started))
3. Create a codespace on that repository and observe all extensions are installed in the remote host as `codespace` extensions

<img width="344" alt="Screenshot 2022-07-08 at 21 55 00" src="https://user-images.githubusercontent.com/61341/178068995-b894f8d6-be57-4a8b-bfce-96417ec9a158.png">

Expected behaviour - the multi-host extension should install as a web extension into `browser` extensions.

When installing this extension manually, it behaves correctly:

<img width="350" alt="Screenshot 2022-07-08 at 21 53 12" src="https://user-images.githubusercontent.com/61341/178068761-042508fe-9e74-4617-ba9a-dc5bfad9f271.png">

**Comment by thomas-sickert** [comment]
Hey @thegecko sorry you're not having the experience you expect. 

I'll file a ticket to discuss this further. In the meantime, if you're able to, it might be worth trying to override the installation location in the [settings.json](https://code.visualstudio.com/docs/devcontainers/containers#_advanced-forcing-an-extension-to-run-locally-or-remotely) file--that could be a potential work around (although I haven't tried it myself).

**Comment by riestmo** [comment]
Hi @thomas-sickert,
I was running into the same problem as @thegecko and unfortunately the suggested workaround does not work for me.
My settings file looks like this:

`{
    "remote.extensionKind": {
        "arm.device-manager": [
            "ui"
        ]
    }
}`

Do you see any other workaround or do I have to wait until the issue is fixed?

**Comment by thegecko** [comment]
I can also confirm this work around doesn't work.

The closest I've got is to use an `extensions.json` file to recommend extensions the user installs. This still means the extensions don't appear automatically, but do end up in the right host if the user installs them.

@thomas-sickert is there any progress on fixing this or a way I can escalate it?

**Comment by thegecko** [comment]
This is still an issue and VS Code extensions marked as `ui` are getting installed in the codespaces host.

Any update @thomas-sickert ?
